mjpeg-view/src/main/java/com/github/niqdev/mjpeg/HeaderExtractor.java

```diff
@@ -29,7 +29,7 @@ static int parseContentLength(byte[] headerBytes) throws IOException, IllegalArg
         int end = i;
 
         // UNTIL the \n at the end of the line
-        while(headerBytes[end] != '\n') {
+        while(headerBytes[end] != '\n' && headerBytes[end] != '\r') {
```

Comment:
interesting catch - maybe instead of the character comparison you could add a private helper function `isNewlineOrCR` (please check if java has a method for that already)

Reply:
thanks, i tried looking but couldnt find anything that Java provides, i've pushed the code now but will make a change to this if i do find anything later on :)